src/test/java/technology/dice/dicefairlink/driver/AuroraReadReplicasDriverConnectAlternatingListTest.java

```diff
@@ -60,6 +63,7 @@
   public void canConnectToValidUrlBasicAuth_thenListOfReplicasChanges() throws Exception {
```

Comment:
test is far too long and near impossible toi understand what is going on. We should either break into small clearly described steps, or create a series of simpler tests that check each item from this test individually.

Reply:
The key element of the test is this mock:
```
    Mockito.when(mockDbCluster.getDBClusterMembers())
        .thenReturn(Arrays.asList(mockDbClusterMember_A, mockDbClusterMember_B))
        .thenReturn(Arrays.asList(mockDbClusterMember_A, mockDbClusterMember_C))
        .thenReturn(Arrays.asList(mockDbClusterMember_B, mockDbClusterMember_C));
```

The test itself is very important as it verifies that when list of cluster members changes over time the driver behaves as expected and distributes load evenly.

Given fact that we use AWS client lib directly, I see no other ways to isolate the driver logic and create fully deterministic configuration for the test to run in.

Any suggestion how to simplify the test are welcome.

Reply:
The key element of the test is this mock:
```
    Mockito.when(mockDbCluster.getDBClusterMembers())
        .thenReturn(Arrays.asList(mockDbClusterMember_A, mockDbClusterMember_B))
        .thenReturn(Arrays.asList(mockDbClusterMember_A, mockDbClusterMember_C))
        .thenReturn(Arrays.asList(mockDbClusterMember_B, mockDbClusterMember_C));
```

The test itself is very important as it verifies that when list of cluster members changes over time the driver behaves as expected and distributes load evenly.

Given fact that we use AWS client lib directly, I see no other ways to isolate the driver logic and create fully deterministic configuration for the test to run in.

Any suggestion how to simplify the test are welcome.